templates/common/infra/bicep/core/search/search-services.bicep

```diff
@@ -0,0 +1,41 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param sku object = {
+  name: 'standard'
+}
+
+param authOptions object = {}
+param semanticSearch string = 'disabled'
+
+resource search 'Microsoft.Search/searchServices@2021-04-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    authOptions: authOptions
+    disableLocalAuth: false
+    disabledDataExfiltrationOptions: []
+    encryptionWithCmk: {
+      enforcement: 'Unspecified'
+    }
+    hostingMode: 'default'
+    networkRuleSet: {
+      bypass: 'None'
+      ipRules: []
+    }
+    partitionCount: 1
+    publicNetworkAccess: 'Enabled'
+    replicaCount: 1
+    semanticSearch: semanticSearch
```

Comment:
Consider adding some additional params with intelligent defaults for some of the more commonly used properties.

Reply:
Done. thanks